rpc/src/client.rs

```diff
@@ -271,6 +271,14 @@ pub trait Client {
         Ok(self.perform(genesis::Request::default()).await?.genesis)
     }
 
+    /// `/genesis_chunked`: get genesis file in multiple chunks.
+    async fn genesis_chunked(
```

Comment:
It would also be nice if we could define a default method to retrieve all parts as once as a [`Stream`](https://docs.rs/futures/latest/futures/stream/trait.Stream.html).

eg. we could add another method with this signature:

```rust
async fn genesis_chunked_stream(&self) -> impl Stream<Item = Result<genesis_chunked::Response, Error>> {
  // ...
}
```

Remains to see exactly how to implement this.

Reply:
Perhaps something like this?

```rust
    async fn genesis_chunked_stream(&self) -> Box<dyn Stream<Item = Result<Vec<u8>, Error>> + '_> {
        Box::new(futures::stream::unfold(Some(0), move |chunk| async move {
            let Some(chunk) = chunk else {
                // An error occurred, abort the stream
                return None;
            };

            match self.genesis_chunked(Some(chunk)).await {
                Ok(response) => {
                    if response.chunk >= response.total {
                        // No more chunks to fetch
                        None
                    } else {
                        // Emit this chunk and fetch the next chunk
                        Some((Ok(response.data), Some(response.chunk + 1)))
                    }
                },
                Err(e) => Some((Err(e), None)), // Abort the stream
            }
        }))
    }
```